integration-tests/tests/src/tests/enums.ts

```diff
@@ -0,0 +1,87 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import {
+  ConnectionState,
+  NumericLogLevel,
+  OpenRealmBehaviorType,
+  OpenRealmTimeOutBehavior,
+  SessionStopPolicy,
+} from "realm";
+
+describe("Enums", function () {
+  describe("ConnectionState", function () {
+    it("is accessible", function () {
+      expect(ConnectionState).to.deep.equal({
+        Disconnected: "disconnected",
+        Connecting: "connecting",
+        Connected: "connected",
+      });
+    });
+  });
+  describe("SessionStopPolicy", function () {
+    it("is accessible", function () {
+      expect(SessionStopPolicy).to.deep.equal({
+        AfterUpload: "after-upload",
+        Immediately: "immediately",
+        Never: "never",
+      });
+    });
+  });
+  describe("OpenRealmBehaviorType", function () {
+    it("is accessible", function () {
+      expect(OpenRealmBehaviorType).to.deep.equal({
+        DownloadBeforeOpen: "downloadBeforeOpen",
+        OpenImmediately: "openImmediately",
+      });
+    });
+  });
+  describe("OpenRealmTimeOutBehavior", function () {
+    it("is accessible", function () {
+      expect(OpenRealmTimeOutBehavior).to.deep.equal({
+        OpenLocalRealm: "openLocalRealm",
+        ThrowException: "throwException",
+      });
+    });
+  });
+  describe("NumericLogLevel", function () {
+    it("is accessible", function () {
+      expect(NumericLogLevel).to.deep.equal({
```

Comment:
Just checked and we do use `deep.equals` 161 trimes and `to.deep.equal` 9 times... with that in mind I am leaning towards being consistent... sorry for leading you wrong way

Reply:
I like consistency so I appreciate that you lifted this :D